src/PCHIPInterpolation.jl

```diff
@@ -22,7 +22,7 @@ function _pchip_ds_scipy(xs::AbstractVector, ys::AbstractVector)
     Δ(i) = (ys[i+1] - ys[i]) / h(i)
 
     length(ys) != length(xs) && throw(DimensionMismatch)
-    ds = similar(ys ./ xs)
```

Comment:
Actually, this division is there to support arrays with physical dimensions (e.g. `Unitful`) where otherwise the units of `ds` would be wrong. I'd suggest keeping it unless there's a better way to do it (which we should probably do in a new PR)

Reply:
Ah, sorry about that. Overall, I think there are opportunities to refactor, but yes I should do that in another PR